diff -urp lignumCAD.pld/gl2ps.h lignumCAD.pld2/gl2ps.h
--- lignumCAD.pld/gl2ps.h	2002-06-14 19:52:59.000000000 +0200
+++ lignumCAD.pld2/gl2ps.h	2010-09-15 20:03:48.000000000 +0200
@@ -213,7 +213,7 @@ GLvoid gl2psBeginPage(GLint format,
 		      GLint colormode, GLint colorsize, GL2PSrgba *colormap, 
 		      GLint buffersize, void * stream);
 #endif
-GLint  gl2psEndPage(GLvoid);
+GLint  gl2psEndPage(void);
 GLvoid gl2psText(char *str, char *fontname, GLint size);
 GLvoid gl2psEnable(GLint mode);
 GLvoid gl2psDisable(GLint mode);
@@ -222,9 +222,9 @@ GLvoid gl2psLineWidth(GLfloat value);
 GLvoid gl2psLineStipple(GLuint mode);
 
 #ifdef GL2PS_USE_QT
-GLvoid gl2psPrintQtHeader ( GLvoid );
+GLvoid gl2psPrintQtHeader ( void );
 GLvoid gl2psPrintQtPrimitive ( GLvoid* a, GLvoid* b );
-GLvoid gl2psPrintQtFooter ( GLvoid );
+GLvoid gl2psPrintQtFooter ( void );
 #endif
 
 #ifdef __cplusplus
diff -urp lignumCAD.pld/gl2ps_qt.cpp lignumCAD.pld2/gl2ps_qt.cpp
--- lignumCAD.pld/gl2ps_qt.cpp	2010-09-15 19:08:46.000000000 +0200
+++ lignumCAD.pld2/gl2ps_qt.cpp	2010-09-15 20:13:39.000000000 +0200
@@ -98,7 +98,7 @@ extern "C" {
    *       Screen Coord[DOTS] * Printer window[DOTS] = Printer Coord[DOTS]
    *     This also takes care of the aspect ratio of the printer window.
    */
-  GLvoid gl2psPrintQtHeader ( GLvoid )
+  GLvoid gl2psPrintQtHeader ( void )
   {
     painter = reinterpret_cast<QPainter*>( gl2ps.stream );
 
@@ -206,7 +206,7 @@ extern "C" {
     }
   }
 
-  GLvoid gl2psPrintQtFooter ( GLvoid )
+  GLvoid gl2psPrintQtFooter ( void )
   {
     painter->restore();
   }
