diff -urp lignumCAD.pld/openglbase.h lignumCAD.pld2/openglbase.h
--- lignumCAD.pld/openglbase.h	2002-09-11 22:53:07.000000000 +0200
+++ lignumCAD.pld2/openglbase.h	2010-09-15 19:36:16.000000000 +0200
@@ -383,7 +383,7 @@ private:
    * \param a value to convert.
    * \return smallest power of two which is greater than or equal to a.
    */
-  uint OpenGLBase::nearestPowerCeil ( unsigned int a );
+  uint nearestPowerCeil ( unsigned int a );
 };
 
 /*!
diff -urp lignumCAD.pld/openglview.cpp lignumCAD.pld2/openglview.cpp
--- lignumCAD.pld/openglview.cpp	2002-12-04 15:20:48.000000000 +0100
+++ lignumCAD.pld2/openglview.cpp	2010-09-15 19:58:14.000000000 +0200
@@ -53,7 +53,7 @@ public:
   /*!
    * Constructor takes the usual arguments defining the old and new state.
    */
-  ChangeViewCommand::ChangeViewCommand ( const QString& name,
+  ChangeViewCommand ( const QString& name,
 					 const PageView* page_view,
 					 const ViewData& old_view,
 					 const ViewData& new_view )
diff -urp lignumCAD.pld/annotationview.cpp lignumCAD.pld2/annotationview.cpp
--- lignumCAD.pld/annotationview.cpp	2010-09-15 19:08:46.000000000 +0200
+++ lignumCAD.pld2/annotationview.cpp	2010-09-15 20:50:18.000000000 +0200
@@ -375,7 +375,7 @@ namespace Space2D {
      * \param new_origin the new origin of the annotation (in model units)
      * \param new_size the new size of the annotation box (in paper units)
      */
-    MoveAnnotationCommand::MoveAnnotationCommand ( const QString& name,
+    MoveAnnotationCommand ( const QString& name,
 						   DesignBookView* design_book_view,
 						   const DBURL& db_url,
 						   Point old_origin,
diff -urp lignumCAD.pld/ocsolidview.cpp lignumCAD.pld2/ocsolidview.cpp
--- lignumCAD.pld/ocsolidview.cpp	2002-12-13 17:17:40.000000000 +0100
+++ lignumCAD.pld2/ocsolidview.cpp	2010-09-15 21:29:31.000000000 +0200
@@ -231,7 +231,7 @@ namespace Space3D {
      * \param solid reference to solid.
      * \param old_parameters old parameters of solid.
      */
-    ModifySolidCommand::ModifySolidCommand ( const QString& name,
+    ModifySolidCommand ( const QString& name,
 					     DesignBookView* design_book_view,
 					     OCSolid* solid,
 					     const std::map<QString,double>&
